spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/types/offer/ContractOffer.java

```diff
@@ -192,12 +192,11 @@ public Builder policy(Policy policy) {
             return this;
         }
 
-
         public ContractOffer build() {
             Objects.requireNonNull(contractOffer.id);
             Objects.requireNonNull(contractOffer.asset, "Asset must not be null");
             Objects.requireNonNull(contractOffer.policy, "Policy must not be null");
-
+            Objects.requireNonNull(contractOffer.contractEnd, "ContractEnd must not be null");
```

Comment:
I have a question... is it correct to have the `contractEnd` on the `ContractOffer`? 
I think this is something that we inherited from the ids infomodel, but on the offer there should be reported only the duration, not the end date, because the offer could be negotiated some time later. 
I'd say the the end date should be calculated when the `ContractAgreement` is created. 
But maybe this is a discussion that needs to be brought to an higher level, I'd ask also to @juliapampus and @jimmarino in relation with the new protocol.

Reply:
We have been looking at fixing this in the IDS spec. My view is that both the end and duration should be specified as policy constraints, and only a timestamp should exist on the agreement.

Reply:
As discussed, we keep the current fix minimal in terms of changes so that we can migrate easily to the policy-approach when new IDS protocol is ready ;)